src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -325,6 +326,38 @@ export default class SourceFileLinter {
 				messageDetails: this.#messageDetails ? formatMessage(MESSAGES.DETAILS__LIB_INIT_2) : undefined,
 			});
 		}
+
+		if (initArg) {
+			this._libInitDeprecatedLibsCheck(initArg);
+		}
+	}
+
+	_libInitDeprecatedLibsCheck(initArg: ts.ObjectLiteralExpression) {
+		const dependenciesNode = initArg.properties.find((prop) => {
+			return ts.isPropertyAssignment(prop) &&
+				ts.isIdentifier(prop.name) &&
+				prop.name.text === "dependencies";
+		});
+
+		if (!dependenciesNode ||
+			!ts.isPropertyAssignment(dependenciesNode) ||
+			!ts.isArrayLiteralExpression(dependenciesNode.initializer)) {
+			return;
+		}
+
+		dependenciesNode.initializer.elements.forEach((dependency) => {
+			const curLibName = dependency.getText().replace(/"/gi, "");
```

Comment:
Done

Reply:
Actually, the `text` property is a bit missleading. The `getText()`. Basically, `getText()` pseudo repsects types. So, the double quotes escaping is stating just that this is a string. Of course, it's obvious from the Node type

Reply:
But how would it look like if the code uses single quotes? AFAIK it would give you also single-quotes in `getText()`.
And if there wouldn't be a string, but just `sap.ui.commons`, it would have still matched with the previous logic, which would be wrong.

This is why there should be a proper type check so that we can just inspect the string content in here, which is the only part we're interested in.

The solution looks a bit complicated (why checking `deprecatedLibraries.includes(curLibName)` when `curLibName` is empty?), but it does the job.

Reply:
done